sw/device/tests/sim_dv/keymgr_key_derivation.c

```diff
@@ -0,0 +1,333 @@
+    0x4e919d54, 0x322288d8, 0x4bd127c7, 0x9f89bc56,
+    0xb4fb0fdf, 0x1ca1567b, 0x13a0e876, 0xa6521d8f};
+
+/**
+ * Key manager Owner Secret stored in info flash page.
+ */
+const uint32_t kOwnerSecret[kSecretWordSize] = {
+    0xa6521d8f, 0x13a0e876, 0x1ca1567b, 0xb4fb0fdf,
+    0x9f89bc56, 0x4bd127c7, 0x322288d8, 0x4e919d54,
+};
+
+/**
+ * Kmac prefix "KMAC" with empty custom string
+ */
+#define KMAC_PREFIX_SIZE 11
+const uint32_t kKmacPrefix[KMAC_PREFIX_SIZE] = {
+    0x4d4b2001, 0x00014341, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+};
+
+/** ROM_EXT key manager maximum version. */
+const uint32_t kMaxVerRomExt = 1;
+
+/** BL0 key manager maximum version. */
+const uint32_t kMaxVerBl0 = 2;
+
+const test_config_t kTestConfig;
+
+/**
+ * Writes `size` words of `data` into flash info page.
+ *
+ * @param page_id Info page ID to write to.
+ * @param data Data to write.
+ * @param size Number of 4B words to write from `data` buffer.
+ */
+static void write_info_page(uint32_t page_id, const uint32_t *data,
+                            size_t size) {
+  mp_region_t info_region = {.num = page_id,
+                             .base = 0x0,  // only used to calculate bank id.
+                             .size = 0x1,  // unused for info pages.
+                             .part = kInfoPartition,
+                             .rd_en = true,
+                             .prog_en = true,
+                             .erase_en = true,
+                             .scramble_en = false};
+  flash_cfg_region(&info_region);
+
+  uint32_t address = FLASH_MEM_BASE_ADDR + page_id * flash_get_page_size();
+
+  ASSERT_EQZ(flash_page_erase(address, kInfoPartition));
+  ASSERT_EQZ(flash_write(address, kInfoPartition, data, size));
+
+  for (size_t i = 0; i < size; ++i) {
+    uint32_t got_data;
+    ASSERT_EQZ(flash_read(address + i * sizeof(uint32_t), kInfoPartition,
+                          /*size=*/1, &got_data));
+    CHECK(got_data == data[i]);
+  }
+  info_region.rd_en = false;
+  info_region.prog_en = false;
+  info_region.erase_en = false;
+  flash_cfg_region(&info_region);
+}
+
+static void init_flash(void) {
+  // setup default access for data partition
+  flash_default_region_access(/*rd_en=*/true, /*prog_en=*/true,
+                              /*erase_en=*/true);
+  // Initialize flash secrets.
+  write_info_page(kFlashInfoPageIdCreatorSecret, kCreatorSecret,
+                  ARRAYSIZE(kCreatorSecret));
+  write_info_page(kFlashInfoPageIdOwnerSecret, kOwnerSecret,
+                  ARRAYSIZE(kOwnerSecret));
+}
+
+static const dif_pwrmgr_wakeup_reason_t kWakeUpReasonPor = {
+    .types = 0,
+    .request_sources = 0,
+};
+
+/** Check wakeup reason to determine what to do during the test */
+static bool compare_wakeup_reasons(const dif_pwrmgr_wakeup_reason_t *lhs,
+                                   const dif_pwrmgr_wakeup_reason_t *rhs) {
+  return lhs->types == rhs->types &&
+         lhs->request_sources == rhs->request_sources;
+}
```

Comment:
IMO this testutil can be reused:
https://github.com/lowRISC/opentitan/blob/18cde68b2f3689f79a320f3006f84f06d33513e1/sw/device/lib/testing/pwrmgr_testutils.c#L27

Reply:
changed to use `dif_rstmgr_software_device_reset`.